sql/src/main/java/com/amazon/opendistroforelasticsearch/sql/sql/parser/AstAggregationBuilder.java

```diff
@@ -136,7 +137,8 @@ private boolean isAggregatorNotFoundAnywhere() {
 
   private boolean isNonLiteralFunction(UnresolvedExpression expr) {
     // The base case for recursion
-    if (expr instanceof Literal) {
+    // Interval is included since it is a special case of literal
+    if (expr instanceof Literal || expr instanceof Interval) {
```

Comment:
Could you try out the workaround here? https://github.com/opendistro-for-elasticsearch/sql/issues/991#issuecomment-760542246

Because I realized what we really want to check here is **if any qualified name (field) not in aggregate function call**. But here we're checking it in the opposite way which may miss some cases. For example, CAST is not covered.

Reply:
I see. Changed to the `isNonAggregateOrLiteralExpression` recursion, and added `cast` and `case when` cases to unit test